### ISSUE TITLE
`space-after-keywords` fails with Allman brace style

### ISSUE THREAD

**Issue by wabain** [eslint/eslint#4149]
This is similar to #3226 and has also been described in comments to #4006.

Using ESLint 1.6.0, the `space-after-keywords` rule gives an error on code with Allman style braces because it expects there to be whitespace which is not a newline between an `else` statement and a subsequent brace:

```js
/* eslint space-after-keywords: [2, "always"]  brace-style: [2, "allman"] */

var x = 1;

if (x)
{
   //...
}
else
{
   //...
}

// 11:1 - Keyword "else" must not be followed by a newline. (space-after-keywords)
```

Something similar happens after a `do` in a do-while statement. From the discussion on #3226 and #4006 it seems like the right approach here would be for `space-after-keywords` to allow newlines in these contexts and leave the rest to the `brace-style` rule.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Confirmed in online demo. Agree, I don't think `space-after-keywords` should check for newline (other then to suppress a warning for missing space).

**Comment by nzakas** [comment]
Working on this.

**Comment by nzakas** [comment]
I put together a PR for this, though I realize there might be some edge cases like `function` (where you never want a newline). Just want to be sure that's what we want before making the change.

**Comment by nzakas** [comment]
@ilyavolodin thoughts?

**Comment by ilyavolodin** [comment]
@nzakas Can you elaborate on function example? Do you mean something like this:
```js
function
a() {
...
}
```

I don't think it's part of this rule's job to check for something like this. This rule should only care that a space exists after a given keyword. The only exception should be that if keyword is followed by line-break, space is not necessary.

**Comment by nzakas** [comment]
Yup, that's what I'm talking about. If we're okay with the rule not catching that, then we're good to go.

**Comment by ilyavolodin** [comment]
If I'm not mistaken, the example above will fail to parse anyways. Again, I think we are overloading this rule with functionality if we are trying to make it check anything other then existence of space after a keyword. Rest of problems will be captured by either parser or other rules.

**Comment by nzakas** [comment]
It seems to parse OK. 

You're probably right that this rule is doing too much and it shouldn't check newlines. Removing that seems like a breaking change to me. What do you think?

**Comment by ilyavolodin** [comment]
Hmm... That's a good question. I think it might be breaking. I'm not sure how people are using this rule right now, so it's hard to say. Just to clarify, I think we should still check for new lines, but only to verify that space isn't necessary if the keyword is followed by new line.

**Comment by nzakas** [comment]
Also @mysticatea had proposed that we combine `space-before-keywords` and `space-after-keywords` into `keyword-spacing`, so maybe we just hold off on making changes here and do that?

**Comment by ilyavolodin** [comment]
Sure. That sounds fine to me.

**Comment by PhiLhoSoft** [comment]
Thanks for the PR, @nzakas, at least I could apply it manually on my copy of ESLint... :smile: 
Indeed, this part of code seems not to belong to the rule.
